yt_dlp/extractor/tumblr.py

```diff
@@ -344,44 +545,37 @@ def _real_extract(self, url):
                         'quality': quality,
                     } for quality, (video_url, format_id) in enumerate(sources)]
 
-        if not media_json.get('url') and not video_url and not iframe_url:
-            # external video host (but we weren't able to figure it out from the api)
-            iframe_url = self._search_regex(
-                r'src=["\'](https?://safe\.txmblr\.com/svc/embed/inline/[^"\']+)["\']',
-                webpage, 'embed iframe url', default=None)
-            return self.url_result(iframe_url or redirect_url, 'Generic')
+        if not formats and fallback_format:
+            formats.append(fallback_format)
 
-        formats = formats or [{
-            'url': media_json.get('url') or video_url,
-            'width': int_or_none(
-                media_json.get('width') or self._og_search_property('video:width', webpage, default=None)),
-            'height': int_or_none(
-                media_json.get('height') or self._og_search_property('video:height', webpage, default=None)),
-        }]
+        if formats:
+            # tumblr's own video is always above embeds
+            entries = [{
+                **info_dict,
+                'formats': formats,
+                'duration': duration,
+                'thumbnail': traverse_obj(
+                    video_json, ('poster', 0, 'url')) or self._og_search_thumbnail(
+                        webpage, default=None)},
+                *entries]
 
-        # the url we're extracting from might be an original post or it might be a reblog.
-        # if it's a reblog, og:description will be the reblogger's comment, not the uploader's.
-        # content_json is always the op, so if it exists but has no text, there's no description
-        if content_json:
-            description = '\n\n'.join(
-                item.get('text') for item in content_json if item.get('type') == 'text') or None
-        else:
-            description = self._og_search_description(webpage, default=None)
-        uploader_id = traverse_obj(post_json, 'reblogged_root_name', 'blog_name')
+        if ignored_providers:
+            if not entries:
+                raise ExtractorError(f'None of embed providers are supported: {", ".join(ignored_providers)!s}', video_id=video_id, expected=True)
+            else:
+                self.report_warning(f'Skipped embeds from unsupported providers: {", ".join(ignored_providers)!s}', video_id)
+        if unknown_providers:
+            self.report_warning(f'Unrecognized providers, please report: {", ".join(unknown_providers)!s}', video_id)
 
+        if not entries:
+            raise ExtractorError('No video could be found in this post', expected=True)
```

Comment:
since we do have metadata to return at this point, we could allow `--ignore-no-formats`
```suggestion
            self.raise_no_formats('No video could be found in this post', expected=True, video_id=video_id)
```